diff --git a/CMakeLists.txt b/CMakeLists.txt
index 136d369..06a090a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,12 +57,7 @@ pkg_check_modules(Jansson REQUIRED jansson)
 include(${CMAKE_ROOT}/Modules/ExternalProject.cmake)
 
 # libhandlegraph (full build using its cmake config)
-ExternalProject_Add(handlegraph
-  SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/deps/libhandlegraph"
-  CMAKE_ARGS "${CMAKE_ARGS};-DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>")
-ExternalProject_Get_property(handlegraph INSTALL_DIR)
-set(handlegraph_INCLUDE "${INSTALL_DIR}/include")
-set(handlegraph_LIB "${INSTALL_DIR}/lib")
+find_package(libhandlegraph CONFIG REQUIRED)
 
 if (CMAKE_MAJOR_VERSION EQUAL "3" AND (CMAKE_MINOR_VERSION EQUAL "10" OR CMAKE_MINOR_VERSION EQUAL "11"))
     # Set link directories. We can't yet use target_link_directories to keep
@@ -104,8 +99,8 @@ set_property(TARGET vgio_static PROPERTY POSITION_INDEPENDENT_CODE OFF)
 # Don't build any object files until the Protobuf include symlink is set up and handlegraph is built
 add_dependencies(vgio link_target)
 add_dependencies(vgio_static link_target)
-add_dependencies(vgio handlegraph)
-add_dependencies(vgio_static handlegraph)
+add_dependencies(vgio handlegraph::handlegraph_shared)
+add_dependencies(vgio_static handlegraph::handlegraph_static)
 
 # Add an alias so that library can be used inside the build tree, e.g. when testing
 add_library(VGio::vgio ALIAS vgio)
@@ -119,7 +114,7 @@ target_include_directories(vgio
         ${HTSlib_INCLUDEDIR}
         ${Protobuf_INCLUDE_DIRS}
         ${Jansson_INCLUDEDIR}
-        $<BUILD_INTERFACE:${handlegraph_INCLUDE}>
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/handlegraph::handlegraph_shared>
     PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
@@ -131,7 +126,7 @@ target_include_directories(vgio_static
         ${HTSlib_INCLUDEDIR}
         ${Protobuf_INCLUDE_DIRS}
         ${Jansson_INCLUDEDIR}
-        $<BUILD_INTERFACE:${handlegraph_INCLUDE}>
+        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/handlegraph::handlegraph_static>
     PRIVATE
         ${CMAKE_CURRENT_SOURCE_DIR}/src
 )
@@ -144,11 +139,11 @@ target_compile_features(vgio_static PUBLIC cxx_std_14)
 # Also note that target_link_directories needs cmake 3.13+
 target_link_libraries(vgio
     PUBLIC
-        ${Protobuf_LIBRARIES} Threads::Threads ${HTSlib_LIBRARIES} ${Jansson_LIBRARIES} ${handlegraph_LIB}/libhandlegraph${CMAKE_SHARED_LIBRARY_SUFFIX} ${PLATFORM_EXTRA_LIB_FLAGS} OpenMP::OpenMP_CXX
+        ${Protobuf_LIBRARIES} Threads::Threads ${HTSlib_LIBRARIES} ${Jansson_LIBRARIES} handlegraph::handlegraph_shared ${PLATFORM_EXTRA_LIB_FLAGS} OpenMP::OpenMP_CXX
 )
 target_link_libraries(vgio_static
     PUBLIC
-        ${Protobuf_STATIC_LIBRARIES} Threads::Threads ${HTSlib_STATIC_LIBRARIES} ${Jansson_LIBRARIES} ${handlegraph_LIB}/libhandlegraph${CMAKE_STATIC_LIBRARY_SUFFIX} ${PLATFORM_EXTRA_LIB_FLAGS} OpenMP::OpenMP_CXX
+        ${Protobuf_STATIC_LIBRARIES} Threads::Threads ${HTSlib_STATIC_LIBRARIES} ${Jansson_LIBRARIES} handlegraph::handlegraph_static ${PLATFORM_EXTRA_LIB_FLAGS} OpenMP::OpenMP_CXX
 )
 
 if (NOT (CMAKE_MAJOR_VERSION EQUAL "3" AND (CMAKE_MINOR_VERSION EQUAL "10" OR CMAKE_MINOR_VERSION EQUAL "11")))
