diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0708095..f1e2f57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,11 +1,13 @@
 # Specify the minimum version for CMake
 cmake_minimum_required(VERSION 3.10)
 
+# Project's name
+project(libhandlegraph VERSION 0.0.1 LANGUAGES CXX)
+
 # This defines default install directories like "lib"
 include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
 
-# Project's name
-project(libhandlegraph)
 # We build using c++14
 set(CMAKE_CXX_STANDARD 14)
 
@@ -21,13 +23,13 @@ if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
     # an absolute path instead. This will let us actually find the library when
     # we use it as a CMake external project and don't fully install it to any
     # normal lib directory.
-    message("libhandlegraph is root project or external_project")
+    message(STATUS "libhandlegraph is root project or external_project")
     set (CMAKE_MACOSX_RPATH OFF)
 else()
     # We are probably an add_subdirectory. We will expect to be in the root
     # project's lib directory, so we do want to have our non-installed
     # install_name use @rpath.
-    message("libhandlegraph is add_subdirectory project")
+    message(STATUS "libhandlegraph is add_subdirectory project")
     set (CMAKE_MACOSX_RPATH ON)
 endif()
 
@@ -116,61 +118,42 @@ set_target_properties(handlegraph_objs PROPERTIES POSITION_INDEPENDENT_CODE TRUE
 # installation.
 add_library(handlegraph_shared SHARED $<TARGET_OBJECTS:handlegraph_objs>)
 set_target_properties(handlegraph_shared PROPERTIES OUTPUT_NAME handlegraph)
-target_include_directories(handlegraph_shared INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/include> $<INSTALL_INTERFACE:include>)
+target_include_directories(handlegraph_shared
+	INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/include>;$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>)
 add_library(handlegraph_static STATIC $<TARGET_OBJECTS:handlegraph_objs>)
 set_target_properties(handlegraph_static PROPERTIES OUTPUT_NAME handlegraph)
-target_include_directories(handlegraph_static INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/include> $<INSTALL_INTERFACE:include>)
+target_include_directories(handlegraph_static
+	INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/include>;$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>)
+add_library(handlegraph ALIAS handlegraph_shared)
 
 # Set up for installability
+# Creating the package configuration file
+configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/libhandlegraph-config.cmake.in
+	"${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph-config.cmake"
+	INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libhandlegraph)
+# Generating the version file for the configuration file
+write_basic_package_version_file(
+	"${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph-config-version.cmake"
+	VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}"
+	COMPATIBILITY AnyNewerVersion)
+# Install generated configuration files
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph-config.cmake"
+	"${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph-config-version.cmake"
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libhandlegraph)
+
 # Make sure to put all the targets in an export set
-install(TARGETS handlegraph_shared handlegraph_static EXPORT libhandlegraphTargets 
+install(TARGETS handlegraph_shared handlegraph_static
+  EXPORT libhandlegraph-targets
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(EXPORT libhandlegraph-targets
+	NAMESPACE handlegraph::
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libhandlegraph)
 # We still have to also say to actually install the headers; install(INCLUDES DESTINATION) doesn't do it.
 # TODO: Use file sets here when we can use CMake 3.23+
 install(DIRECTORY src/include/handlegraph
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   FILES_MATCHING PATTERN "*.hpp"
   )
-
-# Set up CMake config files to be installed so that find_package(libhandlegraph) works.
-include(CMakePackageConfigHelpers)
-write_basic_package_version_file(
-  "${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph/libhandlegraphConfigVersion.cmake"
-  VERSION 1.0
-  COMPATIBILITY AnyNewerVersion
-)
-
-# Write all the targets we make, to be imported in a libhandlegraph:: namespace
-# when the package is found.
-export(EXPORT libhandlegraphTargets
-  FILE "${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph/libhandlegraphTargets.cmake"
-  NAMESPACE libhandlegraph::
-)
-# We don't use configure_package_config_file because we *only* provide targets, not any variables.
-configure_file(cmake/libhandlegraphConfig.cmake
-  "${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph/libhandlegraphConfig.cmake"
-  COPYONLY
-)
-
-set(ConfigPackageLocation lib/cmake/libhandlegraph)
-# Make an exports file that gets installed and lists all the targets from the export set
-install(EXPORT libhandlegraphTargets
-  FILE
-    libhandlegraphTargets.cmake
-  NAMESPACE
-    libhandlegraph::
-  DESTINATION
-    ${ConfigPackageLocation}
-)
-install(
-  FILES
-    cmake/libhandlegraphConfig.cmake
-    "${CMAKE_CURRENT_BINARY_DIR}/libhandlegraph/libhandlegraphConfigVersion.cmake"
-  DESTINATION
-    ${ConfigPackageLocation}
-  COMPONENT
-    Devel
-)
